### ISSUE TITLE
Allow associating the payment / Stripe costumer ID for another user instead of the one logged-in

### ISSUE THREAD

**Issue by webdados** [woocommerce/woocommerce-gateway-stripe#1505]
There might be scenarios (like phone order) where the order is being inserted on WooCommerce by a shop owner or agent on behalf of another client (WordPress user).

For example, by using the Shop as Client plugin, that will associate the order to the final customer profile instead of the currently logged-in user.

Because `WC_Stripe_Payment_Gateway::prepare_source` is always called with `get_current_user_id()` it's impossible to associate the payment and user connection on Stripe with the desired user.

As far as I was able to investigate on the plugin code, this method is always called in places where the order object or id is available, so why not just get the user id from the order instead of getting the logged-in one? Are there any security concerns about this?

In short, the desired behavior is:
- Customer wants to make a phone order
- Shop owner inserts the order on the customer behalf, including credit card details (this happens every day on phone orders)
- 3rd party plugin associates the order with the customer account (and creates the account if it's a new client)
- The Stripe payment/customer details should be the ones of the order customer and not of the logged-in user.

**Comment by shendy-a8c** [comment]
Thank you for filing this issue. Some questions:

> Because WC_Stripe_Payment_Request::prepare_source is always called with get_current_user_id() it's impossible to associate the payment and user connection on Stripe with the desired user.

Which part of the code you are referring to? I don't see `WC_Stripe_Payment_Request::prepare_source()` [here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/trunk/includes/payment-methods/class-wc-stripe-payment-request.php).

**Comment by webdados** [comment]
I'm sorry, it's `WC_Stripe_Payment_Gateway::prepare_source()`. I'm going to edit my original ticket and make that correction.

**Comment by nicomollet** [comment]
Having the same issue here.
Auto login the user with the user from the order can be risky.

**Comment by nicomollet** [comment]
Hey @webdados are you still in need of this feature? I made a PR: #2587

**Comment by webdados** [comment]
@nicomollet Yes! Thanks. Let's see if this goes through.

**Comment by webdados** [comment]
@nicomollet any news on this?

Is this PR tested both with the classic and blocks checkout?

**Comment by nicomollet** [comment]
> [@nicomollet](https://github.com/nicomollet) any news on this?
> 
> Is this PR tested both with the classic and blocks checkout?

Hey @webdados it was tested only in classic checkout (not blocks checkout).

The PR didn't make any progress since last year, unfortunately. 
Still using this change in my shop.

**Comment by shendy-a8c** [comment]
@nicomollet I see the PR has been approved. Why not merged?

**Comment by nicomollet** [comment]
> [@nicomollet](https://github.com/nicomollet) I see the PR has been approved. Why not merged?

[The PR](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2587) was approved by a coworker (no real effect), but not by a reviewer with write access.

Automattic still needs to do an action here, if I am not mistaken.
